xmr-btc/src/bob.rs

```diff
@@ -11,31 +16,122 @@ use ecdsa_fun::{
 };
 use rand::{CryptoRng, RngCore};
 use sha2::Sha256;
+use std::convert::{TryFrom, TryInto};
+
+pub mod message;
+pub use message::{Message, Message0, Message1, Message2, Message3, UnexpectedMessage};
+
+pub async fn next_state<
```

Comment:
This could be a function `next` on the actual `State` itself. That would make it consistent with how the individual states are implemented.

Reply:
leaving this for now, I think there is going to be fair bit of churn on this state machine

Reply:
That's a good idea. Let's do this in the next iteration.